source/main/utils/config.js

```diff
@@ -3,20 +3,43 @@ import { readFileSync } from 'fs';
 import yamljs from 'yamljs';
 import type { LauncherConfig } from '../config';
 
+function recurseReplace(obj) {
```

Comment:
I don't love the implementation of this function. I prefer avoiding for loops in favour of maps, which is certainly possible here. 

But its not a big deal. If this solves the issue, and you get the flow issues resolved we can merge this.

If you have issues I'll assist you when I get online.

Reply:
can always improve it in a future PR when things are less busy